model-optimizer/extensions/ops/scatter_test.py

```diff
@@ -0,0 +1,118 @@
+"""
+ Copyright (C) 2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import unittest
+
+import numpy as np
+from generator import generator, generate
+
+from extensions.ops.scatter import ScatterElementsUpdate
+from mo.front.common.partial_infer.utils import int64_array
+from mo.graph.graph import Node
+from mo.utils.unittest.graph import build_graph, regular_op_with_empty_data, result, connect, \
+    valued_const_with_data
+
+
+@generator
+class ScatterElementsInferTest(unittest.TestCase):
+    @generate(*[
+        ([[0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0],
+          [0.0, 0.0, 0.0]],
+         [[1, 0, 2],
+          [0, 2, 1]],
+         [[1.0, 1.1, 1.2],
+          [2.0, 2.1, 2.2]],
+         0,
+         [[2.0, 1.1, 0.0],
+          [1.0, 0.0, 2.2],
+          [0.0, 2.1, 1.2]]),
+
+        ([[1.0, 2.0, 3.0, 4.0, 5.0]],
+         [[1, 3]],
+         [[1.1, 2.1]],
+         1,
+         [[1.0, 1.1, 3.0, 2.1, 5.0]]),
+
+        ([[1.0, 2.0, 3.0, 4.0, 5.0]],
+         [[1, 3]],
+         [[1.1, 2.1]],
+         [1],
+         [[1.0, 1.1, 3.0, 2.1, 5.0]]),
+
+        ([  # 3D case
+          [[1, 2],
+           [3, 4]],
+          [[5, 6],
+           [7, 8]],
+          [[9, 10],
+           [11, 12]]
+        ],
+         [
+          [[1, 0],
+           [0, 1]],
+          [[1, 1],
+           [1, 0]],
+          [[0, 0],
+           [1, 1]]
+         ],
+         [
+             [[0, 0],
```

Comment:
All the elements are 0s. It is not clear how the operation works for this case. Please, make all elements unique and non-zero.